Makefile

```diff
@@ -85,3 +85,11 @@ docker-internal-linkcheck:
 container-internal-linkcheck: link-checker-image-pull
 	$(CONTAINER_RUN) $(CONTAINER_IMAGE) hugo --config config.toml,linkcheck-config.toml --buildFuture
 	$(CONTAINER_ENGINE) run --mount type=bind,source=$(CURDIR),target=/test --rm wjdp/htmltest htmltest
+
+clean-api-reference:
+	rm -rf content/en/docs/reference/kubernetes-api/ref/
+
+api-reference: clean-api-reference ## Build the API reference pages. go needed
+	mkdir content/en/docs/reference/kubernetes-api/ref/
+	cd api-reference && \
```

Comment:
:wave @feloy. Some practical usage questions:
Where/how do you get the correct swagger.json file?
Is there a version number published on the reference pages?
It is my understanding that the spec file for a given release is not "ready" until the release is cut.
How will this be handled in the Make target?
What manual steps will need to be taken to "configure" the generator?

Reply:
The swagger is copied from https://github.com/kubernetes/kubernetes/tree/master/api/openapi-spec

You are right, the swagger.json file for a new Kubernetes release is not ready before the release. We have to wait for this file to be frozen to generate the reference for this new release. How is it handled for the current API reference page? (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/)

I pushed changes to the README to explain the process.

Reply:
Hi @feloy  The existing build script grabs the correct tagged versioned of the spec from a script parameter.
You can look at the build script and Makefile target that does the copying.

I think the website reference build should automatically pull the spec from `kubernetes/kubernetes` and not rely on
a copy from the sub-module (get directly from the source).

As part of the generator or a separate tool (Hugo?), can you retrieve the spec from the `k8s/k8s` repository given a repository name and Git tag?

When should the generation phase be added/triggered to run in the production publishing script?

* The generator changes (identify a new commit/upgrade in the sub-module). The docs team decides to consume the update of the tool.
* The spec changes for a given version (version stays the same, the spec received backwards compatible text updates and `kubernetes/kubernetes` generated a new version of the spec)?
* In coordination with a new release. A new version of the spec has been generated in `kubernetes/kubernetes`.

The current steps are:
Pull in the generator source code and other supporting file(s) as a sub-module.
Remove the entire file sub-tree and recreate the directory, `content/en/docs/reference/kubernetes-api/ref`.
Generate the new sub-tree (with all links resolved) under this path.
Manually check in and merge files into the repository.
The site is rebuilt with the updated reference files.